### PR TITLE
migrate_service_control: Add 3 cases about kill service

### PR DESCRIPTION
**Pull request by cliping**
1. RHEL-130259 - [Migration] Kill dst qemu during Perform Phase of
live migration - p2p migration
2. RHEL7-31564 - [Migration] Kill DST libvirtd during Perform Phase
of live migration - p2p migration
3. RHEL-130061 - [Migration] Kill SRC libvirtd during Perform Phase
of live migration - p2p migration

Signed-off-by: lcheng <lcheng@redhat.com>

### CODE REVIEW

**Comment by cliping** [comment]
Test result:

#avocado run --vt-type libvirt --vt-machine-type q35 virsh.live_migration.kill_service.kill_libvirtd_on_src.non_p2p_live.without_postcopy --vt-connect-uri qemu:///system
WARNING:root:No python imaging library installed. Screendump and Windows guest BSOD detection are disabled. In order to enable it, please install python-imaging or the equivalent for your distro.
WARNING:root:No python imaging library installed. PPM image conversion to JPEG disabled. In order to enable it, please install python-imaging or the equivalent for your distro.
JOB ID     : 035a3b986cfd1f570c3bde7b40568b28b76b7393
JOB LOG    : /root/avocado/job-results/job-2021-09-07T02.53-035a3b9/job.log
 (1/1) type_specific.io-github-autotest-libvirt.virsh.live_migration.kill_service.kill_libvirtd_on_src.non_p2p_live.without_postcopy: PASS (123.81 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 124.36 s

#avocado run --vt-type libvirt --vt-machine-type q35 virsh.live_migration.kill_service.kill_qemu_on_dst.non_p2p_live.without_postcopy --vt-connect-uri qemu:///system
WARNING:root:No python imaging library installed. Screendump and Windows guest BSOD detection are disabled. In order to enable it, please install python-imaging or the equivalent for your distro.
WARNING:root:No python imaging library installed. PPM image conversion to JPEG disabled. In order to enable it, please install python-imaging or the equivalent for your distro.
JOB ID     : 8160c90d95856b14892ccb6505fbb2fe6af38821
JOB LOG    : /root/avocado/job-results/job-2021-09-07T02.55-8160c90/job.log
 (1/1) type_specific.io-github-autotest-libvirt.virsh.live_migration.kill_service.kill_qemu_on_dst.non_p2p_live.without_postcopy: PASS (172.89 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 173.46 s

#avocado run --vt-type libvirt --vt-machine-type q35 virsh.live_migration.kill_service.kill_libvirtd_on_dst.non_p2p_live.without_postcopy --vt-connect-uri qemu:///system
WARNING:root:No python imaging library installed. Screendump and Windows guest BSOD detection are disabled. In order to enable it, please install python-imaging or the equivalent for your distro.
WARNING:root:No python imaging library installed. PPM image conversion to JPEG disabled. In order to enable it, please install python-imaging or the equivalent for your distro.
JOB ID     : ca52d0e3b27aabf7455cd619eacd5c02a7d720ce
JOB LOG    : /root/avocado/job-results/job-2021-09-07T02.58-ca52d0e/job.log
 (1/1) type_specific.io-github-autotest-libvirt.virsh.live_migration.kill_service.kill_libvirtd_on_dst.non_p2p_live.without_postcopy: PASS (173.52 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 174.09 s

**Comment by cliping** [comment]
Test result:

#avocado run --vt-type libvirt --vt-machine-type q35 virsh.migrate_service_control.kill_service --vt-connect-uri qemu:///system
WARNING:root:No python imaging library installed. Screendump and Windows guest BSOD detection are disabled. In order to enable it, please install python-imaging or the equivalent for your distro.
WARNING:root:No python imaging library installed. PPM image conversion to JPEG disabled. In order to enable it, please install python-imaging or the equivalent for your distro.
JOB ID     : 626fe1656d98a264b48b8a9569650d11349d9311
JOB LOG    : /root/avocado/job-results/job-2021-09-15T04.03-626fe16/job.log
 (1/3) type_specific.io-github-autotest-libvirt.virsh.migrate_service_control.kill_service.kill_libvirtd_on_dst.p2p_live.without_postcopy: PASS (170.93 s)
 (2/3) type_specific.io-github-autotest-libvirt.virsh.migrate_service_control.kill_service.kill_qemu_on_dst.p2p_live.without_postcopy: PASS (166.30 s)
 (3/3) type_specific.io-github-autotest-libvirt.virsh.migrate_service_control.kill_service.kill_libvirtd_on_src.p2p_live.without_postcopy: PASS (110.32 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 448.35 s

**Comment by Yingshun** [comment]
conflicted. rebase please.

File: libvirt/tests/src/migration/migrate_service_control.py [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+
+    # For safety reasons, we'd better back up  xmlfile.
+    new_xml = vm_xml.VMXML.new_from_inactive_dumpxml(vm_name)
+    orig_config_xml = new_xml.copy()
+
+    try:
+        # Update guest disk xml
+        libvirt.set_vm_disk(vm, params)
+
+        if not vm.is_alive():
```
Comment (kamilvarga) [inline libvirt/tests/src/migration/migrate_service_control.py:108]: The vm.start is executed at the end of the libvirt.set_vm_disk method. Is this step necessary also here?

**Review by kamilvarga — COMMENTED** [review]
Others LGTM.

**Comment by cliping** [comment]
Depends on: https://github.com/avocado-framework/avocado-vt/pull/3274

**Comment by cliping** [comment]
Test result in modular daemon mode:

#avocado run --vt-type libvirt --vt-machine-type q35 virsh.migrate_service_control.kill_service
No python imaging library installed. Screendump and Windows guest BSOD detection are disabled. In order to enable it, please install python-imaging or the equivalent for your distro.
No python imaging library installed. PPM image conversion to JPEG disabled. In order to enable it, please install python-imaging or the equivalent for your distro.
JOB ID     : 2fc68c818bc22115bf8cab3ccc1a56a4172690ea
JOB LOG    : /root/avocado/job-results/job-2021-11-15T10.22-2fc68c8/job.log
 (1/3) type_specific.io-github-autotest-libvirt.virsh.migrate_service_control.kill_service.kill_libvirtd_on_dst.p2p_live.without_postcopy: PASS (160.58 s)
 (2/3) type_specific.io-github-autotest-libvirt.virsh.migrate_service_control.kill_service.kill_qemu_on_dst.p2p_live.without_postcopy: PASS (143.75 s)
 (3/3) type_specific.io-github-autotest-libvirt.virsh.migrate_service_control.kill_service.kill_libvirtd_on_src.p2p_live.without_postcopy: PASS (90.21 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 395.17 s

File: libvirt/tests/src/migration/migrate_service_control.py (hunk 1 above)
Comment (cliping) [reply libvirt/tests/src/migration/migrate_service_control.py]: Removed it. Thanks.

File: provider/migration/migration_base.py [hunk 2]
```diff
@@ -205,3 +208,32 @@ def poweroff_src_vm(params):
...
+                                               password=server_pwd)
+        cmd = ""
+        if kill_libvirtd_on_dst:
+            cmd = "pkill -9 libvirtd"
+        if kill_qemu_on_dst:
+            cmd = "pkill -9 qemu-kvm"
+        remote.run_remote_cmd(cmd, params, runner_on_target, ignore_status=False)
+
+    if kill_libvirtd_on_src:
+        process.run('kill -9 `pidof libvirtd`', ignore_status=True, shell=True)
```
[the remark this answers is not in the document]
Comment (cliping) [reply provider/migration/migration_base.py]: Updated, thanks.

File: provider/migration/migration_base.py [hunk 3]
```diff
@@ -205,3 +208,32 @@ def poweroff_src_vm(params):
     """
     vm_session = params.get("vm_session")
     vm_session.cmd("poweroff", ignore_all_errors=True)
+
+
+def kill_service(params):
```
[the remark this answers is not in the document]
Comment (cliping) [reply provider/migration/migration_base.py]: Moved it to avocado-vt. Thanks.

**Review by Yingshun — APPROVED** [review]
LGTM

**Review by kamilvarga — APPROVED** [review]
LGTM.

**Pull request merged**